File: app/src/main/java/com/centrefx/idcardinfo/ResultFragment.java
package com.centrefx.idcardinfo;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

import com.centrefx.idcardinfo.model.Result;

import java.io.Serializable;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link ResultFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class ResultFragment extends Fragment {

    private View v;
    private Result result;

    public ResultFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param r A Result object
     * @return A new instance of fragment ResultFragment.
     */
    public static ResultFragment newInstance(Result r) {
        ResultFragment fragment = new ResultFragment();
        Bundle args = new Bundle();
        args.putSerializable("RESULT", r);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            result = (Result) getArguments().getSerializable("RESULT");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        v = inflater.inflate(R.layout.fragment_result, container, false);
        TextView tv1 = initTextView(R.id.tv_num2);
        TextView tv2 = initTextView(R.id.tv_dob2);
        TextView tv3 = initTextView(R.id.tv_gender2);
        TextView tv4 = initTextView(R.id.tv_province2);
        TextView tv5 = initTextView(R.id.tv_civil2);

        tv1.setText("");
        tv2.setText("");
        tv3.setText("");
        tv4.setText("");
        tv5.setText("");

        tv1.setText(result.getIdNum());
        tv2.setText(result.getDateOfBirth());
        tv3.setText(result.getGender());
        tv4.setText(result.getProvince());
        tv5.setText(result.getCivilStatus());
        return v;
    }

    private TextView initTextView(int id) {
        return v.findViewById(id);
    }
}

File: app/src/main/java/com/centrefx/idcardinfo_k/ResultFragment.kt
package com.centrefx.idcardinfo_k

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.centrefx.idcardinfo_k.databinding.FragmentResultBinding
import com.centrefx.idcardinfo_k.model.Result

/**
 * A simple [Fragment] subclass.
 * Use the [ResultFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class ResultFragment : Fragment() {
    private var v: View? = null
    private var result: Result? = null
    private lateinit var binding: FragmentResultBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        result = arguments?.getSerializable("RESULT") as Result?
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        binding = FragmentResultBinding.inflate(inflater, container, false)
        v = binding.root
        val tv1 = binding.tvNum2
        val tv2 = binding.tvDob2
        val tv3 = binding.tvGender2
        val tv4 = binding.tvProvince2
        val tv5 = binding.tvCivil2

        tv1.text = ""
        tv2.text = ""
        tv3.text = ""
        tv4.text = ""
        tv5.text = ""

        tv1.text = result?.idNum
        tv2.text = result?.dateOfBirth
        tv3.text = result?.gender
        tv4.text = result?.province
        if (result?.civilState != null) {
            tv5.text = result?.civilState
        }
        return v
    }

    companion object {
        @JvmStatic
        fun newInstance(r: Result): ResultFragment {
            val fragment = ResultFragment()
            val b = Bundle()
            b.putSerializable("RESULT", r)
            fragment.arguments = b
            return fragment
        }
    }
}